Mathlib/Analysis/SpecialFunctions/Log/ENNReal.lean

```diff
@@ -0,0 +1,231 @@
+/-
+Copyright (c) 2024 Damien Thomine. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Damien Thomine, Pietro Monticone
+-/
+import Mathlib.Analysis.SpecialFunctions.Pow.NNReal
+import Mathlib.Topology.Instances.EReal
+
+/-!
+# Extended nonnegative real logarithm
+
+We define `log` as an extension of the logarithm of a positive real
+to the extended nonnegative reals `ℝ≥0∞`. The function takes values
+in the extended reals `EReal`, with `log 0 = ⊥` and `log ⊤ = ⊤`.
+
+## Main definitions
+- `ENNReal.log`: The extension of the real logarithm to `ℝ≥0∞`.
+- `ENNReal.log_orderIso`, `ENNReal.log_equiv`: `log` seen respectively
+as an order isomorphism and an homeomorphism.
+
+## Main Results
+- `log_strictMono`: `log` is increasing;
+- `ENNReal.log_injective`, `ENNReal.log_surjective`, `ENNReal.log_bijective`: `log` is
+  injective, surjective, and bijective;
+- `ENNReal.log_mul_add`, `ENNReal.log_pow`, `ENNReal.log_rpow`: `log` satisfy
+the identities `log (x * y) = log x + log y` and `log (x ^ y) = y * log x`
+(with either `y ∈ ℕ` or `y ∈ ℝ`).
+
+## TODO
+- Define `exp` on `ℝ≥0∞` and prove its basic properties.
+
+## Tags
+ENNReal, EReal, logarithm
+-/
+namespace ENNReal
+
+/-! ### Definition -/
+section Definition
+
+/-- The logarithm function defined on the extended nonnegative reals `ℝ≥0∞`
+to the extended reals `EReal`. Coincides with the usual logarithm function
+and with `Real.log` on positive reals, and takes values `log 0 = ⊥` and `log ⊤ = ⊤`.
+Conventions about multiplication in `ℝ≥0∞` and addition in `EReal` make the identity
+`log (x * y) = log x + log y` unconditional. -/
+noncomputable def log (x : ℝ≥0∞) : EReal :=
+  if x = 0 then ⊥
+    else if x = ⊤ then ⊤
+    else Real.log (ENNReal.toReal x)
+
+@[simp]
+theorem log_zero : log 0 = ⊥ := by simp [log]
+
+@[simp]
+theorem log_one : log 1 = 0 := by simp [log]
+
+@[simp]
+theorem log_top : log ⊤ = ⊤ := by simp [log]
+
+theorem log_pos_real {x : ℝ≥0∞} (h : x ≠ 0) (h' : x ≠ ⊤) :
+    log x = Real.log (ENNReal.toReal x) := by simp [log, h, h']
+
+theorem log_pos_real' {x : ℝ≥0∞} (h : 0 < x.toReal) :
+    log x = Real.log (ENNReal.toReal x) := by
+  simp [log, Ne.symm (ne_of_lt (ENNReal.toReal_pos_iff.1 h).1),
+    ne_of_lt (ENNReal.toReal_pos_iff.1 h).2]
+
+theorem log_of_nnreal {x : NNReal} (h : x ≠ 0) :
+    log (x : ENNReal) = Real.log x := by simp [log, h]
```

Comment:
```suggestion
theorem log_of_nnreal {x : ℝ≥0} (h : x ≠ 0) :
    log (x : ℝ≥0∞) = Real.log x := by simp [log, h]
```

Reply:
Hi @dupuisf, I've directly pushed your suggested commit but it seems to throw an error.

Unfortunately I am traveling at the moment so I can't access my computer.

Reply:
I reverted.

Reply:
The version by Frédéric is definitely better. You should use it, and open whatever is needed to make the notation work.

Reply:
Ok, I’m going to do it as soon as I can.

Reply:
Right, you need to put `open scoped NNReal` somewhere for this notation to work.

Reply:
Ops, you're right. I'm going to add it later today. 

Thanks.

Reply:
Done.